src/components/molecules/Modal/Modal.jsx

```diff
@@ -0,0 +1,55 @@
+import './Modal.scss';
+import { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from '../../atoms/IconComponent/Icon';
+
+export const Modal = ({ children }) => {
+    const [showModal, setShowModal] = useState(false);
+
+    const keyPress = useCallback(
+        (e) => {
+            if (e.key === 'Escape' && showModal) {
+                setShowModal(false);
+            }
+        },
+        [setShowModal, showModal]
+    );
+
+    const toggleModal = () => {
+        setShowModal((prev) => !prev);
+    };
+
+    useEffect(() => {
+        document.addEventListener('keydown', keyPress);
+        return () => document.removeEventListener('keydown', keyPress);
+    });
```

Comment:
We should pass an empty dependency array in here, to make sure that this useEffect will run only twice, mount and unmount the component

Reply:
I had a warning : "React Hook useEffect has a missing dependency: 'keyPress'. Either include it or remove the dependency array. " so I put a dependency 'keyPress', but if that's a problem, I can always remove this additional 'keyPress' feature from the component.